### PR TITLE
Setup.state as u8 and shorter keys

### PR DESCRIPTION
**Pull request by renzenicolai**

### CODE REVIEW

File: esp32/modules/setup.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ def store_settings(nickname):
 
 def is_developer(nickname):
     if (nickname==""):
-        badge.nvs_set_str('badge', 'setup.state', '2') # Skip the sponsors
+        badge.nvs_set_u8('badge', 'setup.state', 2) # Skip the sponsors
```
Comment (basvs) [inline esp32/modules/setup.py:14]: If we change the type of 'setup.state', we should support backward compatibility / conversion of old values.

**Review by basvs — CHANGES_REQUESTED** [review]
we cannot just change the type of a nvs key without any conversion-code.

**Comment by basvs** [comment]
> badge.nvs_set_str('badge', 'setup.state', '2')

> badge.nvs_get_str('badge', 'setup.state')

'2'

> badge.nvs_get_u8('badge', 'setup.state')

(empty response; "not set")

> badge.nvs_set_u8('badge', 'setup.state', 2)

> badge.nvs_get_u8('badge', 'setup.state')

(empty response; "not set")

> badge.nvs_get_str('badge', 'setup.state')

'2'


We're not ready for type conversion yet. Working on it..

**Comment by basvs** [comment]
Fixed master-branch.

When setting a key, you can now change the key type.

badge.nvs_set_str('badge', 'something', '42')
badge.nvs_get_str('badge', 'something') -> '42'
badge.nvs_get_u8('badge', 'something') -> null

badge.nvs_set_u8('badge', 'something', 42)
badge.nvs_get_str('badge', 'something') -> null
badge.nvs_get_u8('badge', 'something') -> 42

You do **not** need an explicit nvs_erase_key.

**Comment by annejan** [comment]
@basvs since we are re-flashing all . . 
I think we can merge :)

**Review by basvs — APPROVED** [review]

**Comment by basvs** [comment]
agreed

**Pull request merged**